crates/relayer-types/src/core/ics04_channel/packet.rs

```diff
@@ -178,6 +181,32 @@ impl Packet {
 
         height_timed_out || timestamp_timed_out
     }
+
+    pub fn validate_fields(
```

Comment:
Let's add some doc comments to explain what it does, with a couple examples perhaps?

Reply:
Also, the semantics of this function are a bit weird.

Instead of returning the size when it's too big and 0 when it's okay, how about returning a bespoke `Error` type like:

```rust
enum ValidationError {
  MemoTooBig { size: usize, max: usize },
  ReceiverTooBig { size: usize, max: usize },
  Other(Error),
}
```

and returning the actual sizes when all is okay?